lightning/src/ln/peer_handler.rs

```diff
@@ -620,7 +622,7 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 
 												if msg.features.initial_routing_sync() {
 													peer.sync_status = InitSyncTracker::ChannelsSyncing(0);
-													peers.peers_needing_send.insert(peer_descriptor.clone());
+													peer.needing_send = true;
```

Comment:
@TheBlueMatt Could you explain why this needs to be set here?

Reply:
If we need to stream out our routing db, we (obviously) dont copy the whole thing into our outbound send buff (otherwise we'd just OOM ourselves...), so the logic in do_attempt_write_data has to be hit to add messages to the send buffer if the buffer is empty.